app/secrets.py

```diff
@@ -15,8 +13,44 @@
     "EQ_SERVER_SIDE_STORAGE_DATABASE_PASSWORD",
 ]
 
+PURPOSE_AUTHENTICATION = 'authentication'
+PURPOSE_SUBMISSION = 'submission'
+
 
 def validate_required_secrets(secrets):
     for required_secret in EXPECTED_SECRETS:
-        if required_secret not in secrets:
+        if required_secret not in secrets['secrets']:
             raise Exception("Missing Secret [{}]".format(required_secret))
+
+
+class SecretStore:
+    def __init__(self, secrets):
+        if 'secrets' in secrets:
+            self.secrets = secrets['secrets']
+        if 'private_keys' in secrets:
+            self.private_keys = secrets['private_keys']
+        if 'public_keys' in secrets:
+            self.public_keys = secrets['public_keys']
+
+    def get_secret_by_name(self, secret_name):
+        return self.secrets.get(secret_name)
+
+    def get_private_key_by_kid(self, purpose, kid):
+        if purpose in self.private_keys and kid in self.private_keys[purpose]:
+            return jwk.JWK.from_pem(self.private_keys[purpose][kid].encode('utf-8'), 'digitaleq'.encode('utf-8'))
+
+        raise InvalidTokenException("Invalid Private Key Identifier [{}] for Purpose [{}]".format(kid, purpose))
+
+    def get_public_key_by_kid(self, purpose, kid):
+        if purpose in self.public_keys and kid in self.public_keys[purpose]:
+            return jwk.JWK.from_pem(self.public_keys[purpose][kid].encode('utf-8'))
+
+        raise InvalidTokenException("Invalid Public Key Identifier [{}] for Purpose [{}]".format(kid, purpose))
+
+    def get_private_kids_by_purpose(self, purpose):
+        if purpose in self.private_keys:
+            return list(self.private_keys[purpose].keys())
+
+    def get_public_kids_by_purpose(self, purpose):
+        if purpose in self.public_keys:
+            return list(self.public_keys[purpose].keys())
```

Comment:
I also think it might be worth considering the SecretStore returning instances of a Key class, that has properties for kid, purpose, value etc and a method `as_jwk` to return a JWK version of a key.